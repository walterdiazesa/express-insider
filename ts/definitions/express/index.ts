export enum HandlerType {
  ROUTE = "Route",
  MIDDLEWARE = "Middleware",
  ROUTE_MIDDLEWARE = "Route Middleware",
}

export * from './objects';
export * from './verbs';