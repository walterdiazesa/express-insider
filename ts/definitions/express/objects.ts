import { TrailOptions } from "..";
import { HandlerType, Method } from "../../../ts";

export type RouteMatcher = {
  route: TrailOptions['ignoreRoutes'][number]['route'],
  method: Lowercase<TrailOptions['ignoreRoutes'][number]['method']>,
};

export type Route = {
  path: `/${string}`;
  stack: RouteStack[];
  methods: Record<Method, boolean>;
};

type Stack<T extends HandlerType | undefined = undefined> = {
  handle: Function & { stack?: (StackItem | StackItem<HandlerType.ROUTE>)[] };
  name: string;
  params?: undefined;
  path: T extends HandlerType.ROUTE ? string : unknown;
  keys: Record<string, string | boolean | number>[];
  regexp: RegExp;
};

export type StackItem<T extends HandlerType | undefined = undefined> = Stack<T> & {
  route: T extends undefined ? undefined | Route : T extends HandlerType.ROUTE ? Route : undefined;
  // Config params
  ignore: T extends HandlerType.ROUTE ? undefined | true : never;
  showResponse: T extends HandlerType.ROUTE ? undefined | true : never;
  showRequestedURL: T extends HandlerType.ROUTE ? undefined | true : never;
}

export type RouteStack<T extends Method = Method> = Stack & {
  method: T;
};