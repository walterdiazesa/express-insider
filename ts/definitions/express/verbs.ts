export type Method = "get" | "post" | "options" | "all" | "patch" | "delete" | "put" | "head" | "connect" | "trace";