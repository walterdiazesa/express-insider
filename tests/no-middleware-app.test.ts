import request from 'supertest'
import express from 'express'
import { trail } from '../index';

export const noMiddlewareApp = express();

noMiddlewareApp.get('/', (req, res) => {
  res.send({ working: true })
})

trail(noMiddlewareApp, {
  trailId: () => 'test-id',
  timingFormatter: (elapsed) => 0,
  ignoreMiddlewares: true
});

describe('trail with no middleware output', () => {
  it('Should show handler message for GET / route', async () => {
    const log = jest.spyOn(global.console, 'log');
    await request(noMiddlewareApp).get("/");
    expect(log.mock.calls).toMatchSnapshot();
  })
})