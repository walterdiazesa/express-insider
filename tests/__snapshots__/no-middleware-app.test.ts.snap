// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`trail with no middleware output Should show handler message for GET / route 1`] = `
[
  [
    "[test-id] [32mGET / [0m[1mstart[0m",
  ],
  [
    "[test-id] [35mRoute [32mGET / [33m0 ms [32m200 OK[0m",
  ],
  [
    "[test-id] [32mGET / [0m[1mfinish[0m, elapsed time since begin: [33m0 ms[0m",
  ],
]
`;
